### DISCUSSION TITLE
Running theia.cloud without outside internet access after installation

### DISCUSSION THREAD

**Discussion by Deel96** [eclipse-theia/theia-cloud#60] [Q&A]
Hi,
is it possible to run theia.cloud after initial setup in an environment without outside internet access?
What I want to achieve at the end is:
- Running a custom eclipse theia with emf.cloud inside workspaces that run on kubernetes.
Theia.cloud looks very promising as a lightweight way to do this.

The challenges I see are:
- URLs to nip.io (like https://theia.cloud.192.168.39.81.nip.io/)
- Running a local image registry to update the theia images in the future

Is there anything else to watch out for and does someone know how I can solve these challenges?

Best Regards

**Comment by jfaltermeier** [comment]
Hi, 

You don't have to use nip.io for the hostnames. At the moment we are mainly using nip.io because it is easy to get started together with the IP of the nginx-ingress-controller. You may adjust the hostnames via the helm chart. See the `hosts` section. 
Then you may set up the hostname resolution to the nginx-ingress-controller inside your network accordingly. 

Regarding a local docker image registry: I guess you may run this docker registry on your kubernetes cluster as well and only expose it within the cluster itself.

**Comment by Deel96** [comment]
H @jfaltermeier,

Thank you for your answer.
If I understand you correctly I have to change all occurrences of nip.io URLs in this folder https://github.com/eclipsesource/theia-cloud/tree/main/helm/theia.cloud.
After that I would need to setup hostname resolution e.g. via etc/hosts file with the URLs that occur in the helm charts, right?

After my research what I can try for  create entries in etc/hosts

**Reply by jfaltermeier** [reply]
Yes, you can either directly change the values.yaml you want to use or override it when executing helm. For example you can override the host for the landing page with `--set hosts.landing`:
`helm install theia-cloud ./helm/theia.cloud --namespace theiacloud --create-namespace --values ./helm/theia.cloud/valuesMinikube.yaml --set hosts.landing=my.theia.cloud.domain`

Gitlab has a minikube guide that shortly mentions how to switch from nip.io to `/etc/hosts`: https://docs.gitlab.com/charts/development/minikube/#handling-dns
